src/buffer.rs

```diff
@@ -786,4 +828,34 @@ mod test {
         });
         b.bytes = 1000 * 1000 * 3
     }
+
+    #[bench]
+    #[cfg(feature = "benchmarks")]
+    fn bench_image_access_good_cache(b: &mut test::Bencher) {
+        use buffer::{ImageBuffer, Pixel};
+
+        let mut a: RgbImage = ImageBuffer::new(1000, 1000);
+        for mut p in a.pixels_mut() {
+            let rgb = p.channels_mut();
+            rgb[0] = 255;
+            rgb[1] = 23;
+            rgb[2] = 42;
+        }
+
+        b.iter(move || {
+            let image: &RgbImage = test::black_box(&a);
+            let mut sum: usize = 0;
+            for x in 0..1000 {
+                for y in 0..1000 {
```

Comment:
Swapping these two loops makes the access more cache friendly (and thus speeds up the benchmark) but I'm not sure if this way was intentional.

Reply:
Me neither. Maybe we just benchmark both.

Reply:
Without respect to your very likely different cache sizes, roughly double the speed. But the speed difference between previously and now does not seem to matter for row-by-row cache friendly patterns.

Previous:
```
test … bench:   1,560,805 ns/iter (+/- 117,600) = 1922 MB/s
test … bench:     971,162 ns/iter (+/- 42,709) = 3089 MB/s
```

Now:
```
test … bench:   1,770,711 ns/iter (+/- 63,445) = 1694 MB/s
test … bench:     982,404 ns/iter (+/- 45,258) = 3053 MB/s
```

Reply:
That means the performance regression only happens when you iterate the wrong way. Very nice!